ci/lua-types.py

```diff
@@ -0,0 +1,2559 @@
+    out: list[str] = []
+    for i, child in enumerate(el, start=1):
+        if "name" in child.attrib:
+            out.append(child.attrib["name"])
+        else:
+            out.append(f"unk_{i}")
+    return "_or_".join(out)
+
+
+def declare_prefix(name: str, default: str = "") -> str:
+    if name in DF_GLOBAL:
+        return "df.global."
+    if name in DF:
+        return "df."
+    return default
+
+
+def shim_ineritance(data: str) -> str:
+    for line in data.split("\n"):
+        if line.startswith("-- inherit "):
+            name = line.replace("-- inherit ", "").replace("\n", "")
+            data = data.replace(
+                line,
+                f"-- inherited from {name}\n" + shim_ineritance(class_map[name].split("\n", 1)[-1]) + "-- end " + name,
+            )
+    return data
+
+
+def resolve_inheritance(file: Path) -> None:
+    to_replace: dict[str, str] = {}
+    data: str = ""
+    with file.open("r", encoding="utf-8") as src:
+        data = src.read()
+        for m in data.split("\n\n"):
+            if m.startswith("---@class"):
+                to_replace[m] = shim_ineritance(m)
+
+    if to_replace.__len__() > 0:
+        with file.open("r", encoding="utf-8") as src:
+            data = src.read()
+        with file.open("w", encoding="utf-8") as dest:
+            for k in to_replace:
+                data = to_replace[k].join(data.split(k))
+            dest.write(data)
+
+
+def symbols_processing() -> None:
+    print("Processing symbols...")
+    if not Path(PATH_OUTPUT).is_dir():
+        Path(PATH_OUTPUT).mkdir(parents=True, exist_ok=True)
+        with Path(PATH_LIB_CONFIG).open("w", encoding="utf-8") as dest:
+            print(LIB_CONFIG, file=dest)
+        with Path(PATH_CONFIG).open("w", encoding="utf-8") as dest:
+            print(CONFIG, file=dest)
+
+    with Path(f"{PATH_OUTPUT}aliases.lua").open("w", encoding="utf-8") as dest:
+        print(ALIAS, file=dest)
+
+    for file in sorted(Path(PATH_XML).glob("*.xml")):
+        print(f"Symbols -> {file.name}")
+        with Path(f"{PATH_OUTPUT}{file.name.replace('.xml', '.lua')}").open("w", encoding="utf-8") as dest:
+            print(parse_xml(Path(file)), file=dest)
+
+    for file in sorted(Path(PATH_OUTPUT).glob("*.lua")):
+        resolve_inheritance(file)
+
+
+########################################
+#        Signatures processing         #
+########################################
+
+
+PATH_LUAAPI = "./library/LuaApi.cpp"
+PATH_LIBRARY = "./library/"
+PATH_DFHACK_OUTPUT = "./types/library/dfhack.lua"
+
+PATTERN_WRAPM = r"WRAPM\((.+), (.+)\)"
+PATTERN_CWRAP = r"CWRAP\((.+), (.+)\)"
+PATTERN_WRAPN = r"WRAPN\((.+), (.+)\)"
+PATTERN_WRAP = r"WRAP\((.+)\),"
+PATTERN_MODULE_ARRAY = r"dfhack_[\w+_]*module\[\](.|\n)*?NULL,\s{0,1}NULL\s{0,1}\}\n\}"
+PATTERN_LFUNC_ARRAY = r"dfhack_[\w+_]*funcs\[\](.|\n)*?NULL,\s{0,1}NULL\s{0,1}\}\n\}"
+PATTERN_LFUNC_ITEM = r"\{\s\"(\w+)\",\s(\w+)\s\}"
+PATTERN_SIGNATURE = r"^([\w::<>]+).+?\((.*)?\)"
```

Comment:
`[\w::<>]` -- why is the colon listed twice?
shouldn't `(.*)?` be `(.*?)`? alternately, `([^)]*)` might be clearer